[ALLI-6630] Add allowed and disallowed hosts for covers and descripti…

EreMaijala opened the pull request:
…ons.

EreMaijala committed:
[ALLI-6630] Add allowed and disallowed hosts for covers and descriptions.

samuli commented:
- Hmm, toimiiko tämä muuttamatta Loaderin kutsuksi $this->fetchFromAPI:
https://github.com/NatLibFi/NDL-VuFind2/blob/dev/module/Finna/src/Finna/Cover/Loader.php#L241
- Pitäisikö urlit suodattaa jo record-ajurissa? Nyt käyttöliittymään voi päätyä kuva, jonka lataus keskeytetään myöhemmin Loaderissa. Eli pitäisikö toiminto yhdistää url_blocklist:iin? Silloin DNS-lookup kannattanee välimuistittaa (UrlCheckTrait on tällaisenaan tarkoitettu kutsuttavaksi vain asynkronisista toiminnoista?).
- Pitäisikö filtteröinnin koskea myös välimuistista ladattavia urleja?
- config.ini.sample:n korjaukset: 
   - disable > disabled
   - regex-esimerkkiin: "/" ympärille.
   - wild-card -esimerkki  *.bad.host ei taida toimia tässä muodossa?

EreMaijala committed:
Additional fixes.

- Add checks in record drivers.
- Extend tests.
- Improve logging.
- Fix sample config.

EreMaijala committed:
Marc: Avoid url load check with PDFs.

EreMaijala commented:
@samuli

>     * Hmm, toimiiko tämä muuttamatta Loaderin kutsuksi $this->fetchFromAPI:
>       https://github.com/NatLibFi/NDL-VuFind2/blob/dev/module/Finna/src/Finna/Cover/Loader.php#L241

Toimii, se kutsuu loppujen lopuksi processImageURL-metodia.
 
>     * Pitäisikö urlit suodattaa jo record-ajurissa? Nyt käyttöliittymään voi päätyä kuva, jonka lataus keskeytetään myöhemmin Loaderissa. Eli pitäisikö toiminto yhdistää url_blocklist:iin? Silloin DNS-lookup kannattanee välimuistittaa (UrlCheckTrait on tällaisenaan tarkoitettu kutsuttavaksi vain asynkronisista toiminnoista?).

Joo, parempi niin. Lisäsin suodatuksen. Lisäksi muutin niin, että DNS-lookup tehdään vasta, jos kaikki muut kohdat on läpäisty, ja sitä käytetään vain disallowed-listan kanssa. DNS-lookup on edessä joka tapauksessa, jos URL ladataan, joten ei pitäisi haitata, että se tulee vähän etukäteen. Käyttis cachettaa, meidän ei tarvitse huolehtia siitä itse.

>     * Pitäisikö filtteröinnin koskea myös välimuistista ladattavia urleja?

Loaderissa menee turhan monimutkaiseksi, kun välimuistin käsittely on ennen processImageURL:ää, mutta PDF-käsittely vasta sen jälkeen. Mutta nyt kun tarkistus on drivereissa, koskee se tietysti myös jo välimuistissa olevia.
 
>     * config.ini.sample:n korjaukset:
>       
>       * disable > disabled
>       * regex-esimerkkiin: "/" ympärille.
>       * wild-card -esimerkki  *.bad.host ei taida toimia tässä muodossa?

Korjattu.

EreMaijala committed:
Only do checks in record drivers.

EreMaijala committed:
Revert more changes.
